### PR TITLE
offpunk: init at 1.4

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Description of changes

> A command-line and offline-first smolnet browser/feed reader for Gemini, Gopher, Spartan and 
> Web by [Ploum](https://ploum.net/).
> The goal of Offpunk is to be able to synchronize your content once (a day, a week, a month)
> and then browse/organize it while staying disconnected.

https://notabug.org/ploum/offpunk

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 179888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/applications/networking/browsers/offpunk/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{stdenv, testVersion, makeWrapper, python3, lib, fetchFromGitea, timg, less, offpunk, ...}:
+
+let
+  pythonDependencies = (with python3.pkgs; [
```
Comment (AndersonTorres) [inline pkgs/applications/networking/browsers/offpunk/default.nix:4]: Useless parentheses, I believe.

File: pkgs/applications/networking/browsers/offpunk/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    cryptography
+    feedparser
+    pillow
+    readability-lxml
+    requests
+    setproctitle
+  ]);
+  otherDependencies = [ timg less ];
+in
+stdenv.mkDerivation (finalAttrs: {
```
Comment (AndersonTorres) [inline pkgs/applications/networking/browsers/offpunk/default.nix:15]: /me being curious:

This is the new way to get rid of `rec`?

File: pkgs/applications/networking/browsers/offpunk/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    owner = "ploum";
+    repo = "offpunk";
+    rev = "v${finalAttrs.version}";
+    sha256 = "04dzkzsan1cnrslsvfgn1whpwald8xy34wqzvq81hd2mvw9a2n69";
+  };
+
+  buildInputs = [ makeWrapper ] ++ otherDependencies ++ pythonDependencies;
+
+  installPhase = ''
+    install -D ./offpunk.py $out/bin/offpunk
```
Comment (AndersonTorres) [inline pkgs/applications/networking/browsers/offpunk/default.nix:30]: Use the `runHook` decorations.

File: pkgs/applications/networking/browsers/offpunk/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{stdenv, testVersion, makeWrapper, python3, lib, fetchFromGitea, timg, less, offpunk, ...}:
```
Comment (AndersonTorres) [inline pkgs/applications/networking/browsers/offpunk/default.nix:1]: Afair it is not recommended to use variadic listing.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/browsers/offpunk/default.nix (hunk 3 above)
Comment (DamienCassou) [reply pkgs/applications/networking/browsers/offpunk/default.nix:30]: If you meant `runHook preInstall` and `runHook postInstall`, it is now done. If you meant something else, please tell me.

File: pkgs/applications/networking/browsers/offpunk/default.nix (hunk 4 above)
Comment (DamienCassou) [reply pkgs/applications/networking/browsers/offpunk/default.nix:1]: fixed

**Comment by DamienCassou** [comment]
Thank you very much for your feedback @AndersonTorres. I think I took it into account.

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 179888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/applications/networking/browsers/offpunk/default.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/applications/networking/browsers/offpunk/default.nix:32]: Yes, I meant that..

**Comment by AndersonTorres** [comment]
OfBorg is failing - trying to write `homeless-shelter`.

**Comment by DamienCassou** [comment]
@AndersonTorres wrote:
> OfBorg is failing - trying to write homeless-shelter.

yes, this is because of a passthru testVersion I added to the package. The problem is that offpunk 1.4 tried to create configuration files when asking for `--version`. This [has been fixed recently](https://notabug.org/ploum/offpunk/issues/2) but the fix isn't released yet.

I deleted the passthru test.

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 179888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by AndersonTorres** [comment]
This is a bit strange to me not to use the Python packaging framework from Nixpkgs. But it is working, after all.

**Comment by DamienCassou** [comment]
> This is a bit strange to me not to use the Python packaging framework from Nixpkgs.

I agree and I initially tried to use the Python packaging framework. But offpunk comes with one Python file only and no framework configuration. It was easier to bypass the framework than to use it and disable the automatic stuff that weren't compatible. My lack of understanding of Python packaging also played a big role in this decision :-).

**Comment by fabianhjr** [comment]
> error: Function called without required argument "testVersion" at /home/fabian/.cache/nixpkgs-review/pr-180835/nixpkgs/pkgs/applications/networking/browsers/offpunk/default.nix:10

Investigating an issue around a missing argument.

**Comment by fabianhjr** [comment]
Argument seems unused, could be removed.

**Pull request merged**